cassandra/murmur3.py

```diff
@@ -0,0 +1,110 @@
+from six.moves import range
+import struct
+
+
+def body_and_tail(data):
+    l = len(data)
+    nblocks = l // 16
+    tail = l % 16
+    if nblocks:
+        return struct.unpack_from('qq' * nblocks, data), struct.unpack_from('b' * tail, data, -tail), l
+    else:
+        return tuple(), struct.unpack_from('b' * tail, data, -tail), l
+
+
+def rotl64(x, r):
+    mask = 2 ** r - 1
+    rotated = (x << r) | ((x >> 64 - r) & mask)
```

Comment:
You are right. I do the minimal work to make it correct for this use. I will add a note that these are not be general-purpose routines.